base/loading.jl

```diff
@@ -1511,36 +1533,140 @@ function srctext_files(f::IO, srctextpos::Int64)
     return files
 end
 
-# Find the Project.toml that we should load/store to for Preferences
-function get_preferences_project_path(uuid::UUID)
-    for env in load_path()
-        project_file = env_project_file(env)
-        if !isa(project_file, String)
-            continue
-        end
-        if uuid_in_environment(project_file, uuid)
-            return project_file
+# Test to see if this UUID is mentioned in this `Project.toml`; either as
+# the top-level UUID (e.g. that of the project itself) or as a dependency.
+function get_uuid_name(project::Dict, uuid::UUID)
+    if haskey(project, "uuid") && haskey(project, "name") &&
+                                  UUID(project["uuid"]) == uuid
+        return project["name"]
+    elseif haskey(project, "deps")
+        struuid = string(uuid)
+        for (k, v) in project["deps"]
```

Comment:
Similarly here, e.g.

```suggestion
        for (k, v) in project["deps"]
            v::String
```